server/src/main/java/org/opensearch/index/codec/CodecService.java

```diff
@@ -93,6 +93,13 @@ public CodecService(@Nullable MapperService mapperService, IndexSettings indexSe
         this.codecs = codecs.immutableMap();
     }
 
+    /**
+     * Returns default codec
+     */
+    public Codec getDefaultCodec() {
```

Comment:
Please make it `final`. Also, taking the method's naming in this class, `defaultCodec` would be more concise I believe:

```suggestion
    public final Codec defaultCodec() {
```